src/ain/services/crypto.service.ts

```diff
@@ -15,14 +16,23 @@ export enum Blockchain {
 export class CryptoService {
   public verifySignature(message: string, address: string, signature: string): boolean {
     const blockchains = this.getBlockchainsBasedOn(address);
+    const defichainFallbackMessage = Config.auth.signMessage + address;
 
     let isValid = false;
     try {
       isValid = this.verify(message, address, signature, blockchains);
+
+      if (!isValid && blockchains.includes(Blockchain.DEFICHAIN)) {
+        isValid = this.verify(defichainFallbackMessage, address, signature, blockchains);
+      }
     } catch (e) {}
 
     if (!isValid && !blockchains.includes(Blockchain.ETHEREUM)) {
       isValid = this.fallbackVerify(message, address, signature, blockchains);
+
+      if (!isValid && blockchains.includes(Blockchain.DEFICHAIN)) {
+        isValid = this.fallbackVerify(defichainFallbackMessage, address, signature, blockchains);
+      }
```

Comment:
Shouldn't we put this logic into `verifyDefichain` method? Would be easier IMO.

Reply:
yes you are 100% correct. Don't know my reasoning why I did it outside

Reply:
ah because of the already implemented `fallbackVerify` that isn't handled within defichains verify. Hm... In which cases is this `fallbackVerify` executed? Should we in case of Defichain always execute it with the old message?

Reply:
The `fallbackVerify` is used to support all address formats (we got this code from U-Zyn and actually never questioned or verified it 😉). So `fallbackVerify` also needs to be executed for both messages.

Reply:
Okay just tested with a wrong and a valid signature. `fallbackVerify` does use `verify` too and that is deciding which blockchain should be used. Means that both cases have fallback to old message